cmd/utils.go

```diff
@@ -168,104 +173,125 @@ func parseAddress(ctx context.Context, addr string, lapi lotusapi.FullNode) (com
+	opEvm, opFevm, err := as.GetAddrs(util.OperatorKey)
 	if err != nil {
-		return common.Address{}, nil, nil, err
+		return common.Address{}, nil, accounts.Account{}, "", nil, err
 	}
 
-	owEvm, owFevm, err := ks.GetAddrs(util.OwnerKey)
+	owEvm, owFevm, err := as.GetAddrs(util.OwnerKey)
 	if err != nil {
-		return common.Address{}, nil, nil, err
+		return common.Address{}, nil, accounts.Account{}, "", nil, err
 	}
 
-	var pk *ecdsa.PrivateKey
+	var fromAddress common.Address
 	// if no flag was passed, we just use the operator address by default
-	from := cmd.Flag("from").Value.String()
+	// from := cmd.Flag("from").Value.String()
 	switch from {
 	case "", opEvm.String(), opFevm.String():
-		funded, err := as.IsFunded(cmd.Context(), PoolsSDK, opFevm, util.OperatorKeyFunded, opEvm.String())
+		funded, err := as.IsFunded(ctx, PoolsSDK, opFevm, util.OperatorKeyFunded, opEvm.String())
 		if err != nil {
-			return common.Address{}, nil, nil, err
+			return common.Address{}, nil, accounts.Account{}, "", nil, err
 		}
 		if funded {
-			pk, err = ks.GetPrivate(util.OperatorKey)
+			fromAddress = opEvm
 		} else {
 			log.Println("operator not funded, falling back to owner address")
-			pk, err = ks.GetPrivate(util.OwnerKey)
+			fromAddress = owEvm
 		}
 		if err != nil {
-			return common.Address{}, nil, nil, err
+			return common.Address{}, nil, accounts.Account{}, "", nil, err
 		}
 	case owEvm.String(), owFevm.String():
-		pk, err = ks.GetPrivate(util.OwnerKey)
+		fromAddress = owEvm
 	default:
-		return common.Address{}, nil, nil, errors.New("invalid from address")
+		return common.Address{}, nil, accounts.Account{}, "", nil, errors.New("invalid from address")
 	}
 	if err != nil {
-		return common.Address{}, nil, nil, err
+		return common.Address{}, nil, accounts.Account{}, "", nil, err
 	}
 
-	agentAddrStr, err := as.Get("address")
+	agentAddr, err = getAgentAddress()
 	if err != nil {
-		return common.Address{}, nil, nil, err
+		return common.Address{}, nil, accounts.Account{}, "", nil, err
 	}
 
-	if agentAddrStr == "" {
-		return common.Address{}, nil, nil, errors.New("No agent found. Did you forget to create one?")
+	account = accounts.Account{Address: fromAddress}
+	wallet, err = manager.Find(account)
+	if err != nil {
+		return common.Address{}, nil, accounts.Account{}, "", nil, err
 	}
 
-	agentAddr := common.HexToAddress(agentAddrStr)
+	var envSet bool
+	var message string
+	if fromAddress == owEvm {
+		passphrase, envSet = os.LookupEnv("GLIF_OWNER_PASSPHRASE")
+		message = "Owner key passphrase"
+	} else if fromAddress == opEvm {
+		passphrase, envSet = os.LookupEnv("GLIF_OPERATOR_PASSPHRASE")
+		message = "Operator key passphrase"
+	}
+	if !envSet {
+		err = ks.Unlock(account, "")
+		if err != nil {
+			prompt := &survey.Password{Message: message}
+			survey.AskOne(prompt, &passphrase)
+			if passphrase == "" {
+				return common.Address{}, nil, accounts.Account{}, "", nil, fmt.Errorf("Aborted")
+			}
+		}
+	}
```

Comment:
One thought here (which I can test) - if the `operator` has a passphrase, will this break autopilot? Also, will the prompt still exist in the passphrase is `""`?

Reply:
If the passphrase is "", it skips the prompt.

If the operator has a passphrase, it should ask the passphrase each time it creates a transaction if GLIF_OPERATOR_PASSPHRASE isn't set.